frontend/ui/widget/filechooser.lua

```diff
@@ -193,6 +193,16 @@ function FileChooser:genItemTableFromPath(path)
 
             return a.percent_finished < b.percent_finished
         end
+    elseif self.collate == "numeric" then
+        -- adapted from: http://notebook.kulchenko.com/algorithms/alphanumeric-natural-sorting-for-humans-in-lua
+        local function addLeadingZeroes(d)
+            local dec, n = string.match(d, "(%.?)0*(.+)")
+            return #dec > 0 and ("%.12f"):format(d) or ("%s%03d%s"):format(dec, #n, n)
+        end
+        sorting = function(a, b)
+            return tostring(a.name):gsub("%.?%d+", addLeadingZeroes)..("%3d"):format(#b.name)
+         < tostring(b.name):gsub("%.?%d+",addLeadingZeroes)..("%3d"):format(#a.name)
```

Comment:
slightly odd indent here?

Reply:
Indeed, I tried to fix it, but I'm not sure if it's alright. I always had trouble indenting code spanning multiple lines 😅

Reply:
I'd say don't worry too much about line length. :-)

It's mainly an indicator that things could be getting too complex if you're getting super deep loops or something.